core/src/impl/Kokkos_SharedAlloc.hpp

```diff
@@ -24,6 +24,18 @@
 #include <cstdint>
 #include <string>
 
+// Use likely, unlikely attributes from C++20 to improve performance of
+// reference counting when copy constructors are involved.
+#if defined(KOKKOS_ENABLE_CXX20)
+#if defined(KOKKOS_IMPL_REF_COUNT_BRANCH_LIKELY)
+#define KOKKOS_BRANCH_PROB [[likely]]
+#elif defined(KOKKOS_IMPL_REF_COUNT_BRANCH_UNLIKELY)
+#define KOKKOS_BRANCH_PROB [[unlikely]]
```

Comment:
I also noticed that you need another else branch. This was not needed in the original code because it did not have a `KOKKOS_ENABLE_CXX20` branch.
```
#else
#define KOKKOS_BRANCH_PROB
```